lang/rust/rust.py

```diff
@@ -0,0 +1,216 @@
+        actions.auto_insert(' >= ')
+
+    def code_operator_less_than():
+        actions.auto_insert(' < ')
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(' <= ')
+
+    def code_operator_and():
+        actions.auto_insert(' && ')
+
+    def code_operator_or():
+        actions.auto_insert(' || ')
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(' & ')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(' | ')
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(' ^ ')
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(' << ')
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(' <<= ')
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(' >> ')
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(' >>= ')
+
+    def code_import():
+        actions.auto_insert('use ')
+
+    def code_comment():
+        actions.auto_insert('// ')
+
+    def code_self():
+        actions.auto_insert('self')
+
+    def code_operator_object_accessor():
+        actions.auto_insert('.')
+
+    def code_insert_true():
+        actions.auto_insert('true')
+
+    def code_insert_false():
+        actions.auto_insert('false')
+
+    def code_state_if():
+        actions.insert('if {}')
+        for i in range(3):
+            actions.edit.left()
+
+    def code_state_else_if():
+        actions.insert('else if {}')
+        for i in range(3):
+            actions.edit.left()
+
+    def code_state_return():
+        actions.insert("return ")
+
+    def code_insert_function(text: str, selection: str):
+        actions.user.paste(f'fn {text}({selection or ""})')
+        actions.edit.left()
+
+    def code_private_function(text: str):
+        formatter = settings.get('user.code_private_function_formatter')
+        function_name = actions.user.formatted_text(text, formatter)
+        actions.user.code_insert_function(function_name, None)
+
+    def code_default_function(text: str):
+        formatter = settings.get('user.code_private_function_formatter')
+        function_name = actions.user.formatted_text(text, formatter)
+        actions.user.code_insert_function(function_name, None)
+
+    def code_insert_type_annotation(type: str):
+        actions.insert(f': {type}')
+
+    def code_insert_return_type(type: str):
+        actions.insert(f' -> {type}')
+
+    def code_state_for():
+        actions.insert('for ')
+
+    def code_public_function(text: str):
+        actions.insert('pub ')
+        formatter = settings.get('user.code_private_function_formatter')
+        function_name = actions.user.formatted_text(text, formatter)
+        actions.user.code_insert_function(function_name, None)
+
+    def code_state_return():
+        actions.insert('return ')
+
+    def code_insert_is_null():
+        actions.auto_insert(' != None')
```

Comment:
I think you have this the wrong way around.

Also, PartialEq won't be implemented for Option<T> where T doesn't implement PartialEq, right? Isn't there a method which could be used here instead?

Going further, would someone perhaps like to use null and none to refer to the null pointer and the empty Option respectively?

Reply:
Rust doesn’t have null builtin. I’m not sure about what you suggest here.

Reply:
https://doc.rust-lang.org/std/ptr/fn.null.html

Reply:
> https://doc.rust-lang.org/std/ptr/fn.null.html

Yeah, I’ve seen that, but it’s not even in prelude. I don’t think it will be useful in general. It’s just a regular method like any other. `Option` is at least in prelude. And no one is really checking for `None` like that in Rust anyway, so if it would up to me I wouldn’t implemented it at all.